### PR TITLE
feat: transition to TypeScript

### LINKED ISSUE #521: transition to typescript
**Describe the feature**
What feature are you proposing?
ts supremacy

**Priority**
- [x] High
- [ ] Medium
- [ ] Low

**List the benefits of adding such a feature**
What will you gain from this feature?
everything

**Is this feature request related to a problem?**
Is there a problem with the current implementation (if any), or is there an issue that would be resolved with this feature?
N/A

### LINKED ISSUE #524: Quaver while disconnected crashes when a goto page modal is submitted
**Describe the bug**
What isn't working as intended, and what does it affect?
Affects functionality

**Severity**
- [x] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Affected branches**
- [x] Stable
- [x] Next

**Steps to reproduce**
Steps to reproduce the behavior. (e.g. click on a button, enter a value, etc. and see error)
1. Play more than 2 tracks with Quaver
2. Check the queue
3. Click the go to button to open a modal.
4. Restart Quaver or disconnect Quaver
5. If Quaver is started again or is not in any channel, submit the modal

**Expected behavior**
What is expected to happen?
A toast saying that Quaver is not in any channel, or just ignore the error.

**Actual behavior**
What actually happens? Attach or add errors or screenshots here as well.
Quaver will not be able to update the interaction if it there is no player, the modal came from the button, `update` method will not exist.
```js
$ node .
2022-10-07T13:42:28.338Z [Discord] INFO: Connected. Logged in as Aia#3990.     
2022-10-07T13:42:28.342Z [Quaver] INFO: Running version 5.2.3-next.1. For help, see https://github.com/ZPTXDev/Quaver/issues.
2022-10-07T13:42:28.343Z [Quaver] WARN: You are running an unstable version of 
Quaver. Please report bugs using the link above, and note that features may change or be removed entirely prior to release.
2022-10-07T13:42:29.114Z [Lavalink] INFO: Connected.
2022-10-07T13:42:34.556Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing modal queue_goto
2022-10-07T13:42:34.557Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-07T13:42:34.566Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-07T13:42:34.567Z [Quaver] INFO: Shutting down due to discord...        
2022-10-07T13:42:34.569Z [Quaver] ERROR: Cannot read properties of undefined (reading 'components')
TypeError: Cannot read properties of undefined (reading 'components')
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/src/components/modals/queue.js:15:41)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/src/events/interactionCreate.js:114:18)
    at Client.<anonymous> (file:///C:/Users/Ava/Documents/GitHub/Quaver/src/main.js:248:57)
    at Client.emit (node:events:513:28)
    at InteractionCreateAction.handle (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\actions\InteractionCreate.js:81:12)
    at module.exports [as INTERACTION_CREATE] (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\handlers\INTERACTION_CREATE.js:4:36)
    at WebSocketManager.handlePacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketManager.js:352:31)
    at WebSocketShard.onPacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:481:22)
    at WebSocketShard.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:321:10)
    at WebSocket.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\ws\lib\event-target.js:199:18)
2022-10-07T13:42:34.571Z [Quaver] INFO: Logging additional output to error.log.
```

### LINKED ISSUE #527: Quaver with an empty queue crashes when submitting a goto page modal
**Describe the bug**
What isn't working as intended, and what does it affect?
Affects functionality

**Severity**
- [x] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Affected branches**
- [x] Stable
- [x] Next

**Steps to reproduce**
Steps to reproduce the behavior. (e.g. click on a button, enter a value, etc. and see error)

**Expected behavior**
What is expected to happen?

**Actual behavior**
What actually happens? Attach or add errors or screenshots here as well.
```js
$ npm run start

> quaver@5.2.3-next.1 start
> node .

2022-10-08T23:06:07.582Z [Discord] INFO: Connected. Logged in as Aia#3990.
2022-10-08T23:06:07.586Z [Quaver] INFO: Running version 5.2.3-next.1. For help, see https://github.com/ZPTXDev/Quaver/issues.
2022-10-08T23:06:07.587Z [Quaver] WARN: You are running an unstable version of Quaver. Please report bugs using the link above, and note that features may change or be removed entirely prior to release.
2022-10-08T23:06:08.269Z [Lavalink] INFO: Connected.
2022-10-08T23:06:09.568Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command play
2022-10-08T23:06:09.572Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command play
2022-10-08T23:06:12.166Z [Quaver] INFO: [G 906471497231630336] Starting track
2022-10-08T23:06:13.452Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command play
2022-10-08T23:06:13.453Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command play
2022-10-08T23:06:16.342Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command queue
2022-10-08T23:06:16.343Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command queue
2022-10-08T23:06:22.026Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing button queue_goto
2022-10-08T23:06:22.027Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing button queue_goto
2022-10-08T23:06:25.757Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing button queue_goto
2022-10-08T23:06:25.758Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing button queue_goto
2022-10-08T23:06:26.654Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command skip
2022-10-08T23:06:26.655Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command skip
2022-10-08T23:06:26.882Z [Quaver] INFO: [G 906471497231630336] Starting track
2022-10-08T23:06:30.840Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing modal queue_goto
2022-10-08T23:06:30.840Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-08T23:06:30.844Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T23:06:30.845Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T23:06:30.846Z [Quaver] INFO: Disconnecting from all guilds...
2022-10-08T23:06:30.846Z [Quaver] INFO: [G 906471497231630336] Disconnecting (restarting)
2022-10-08T23:06:31.353Z [Quaver] ERROR: Cannot read properties of undefined (reading 'length')
TypeError: Cannot read properties of undefined (reading 'length')
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/components/modals/queue.js:16:38)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/events/interactionCreate.js:125:30)
    at Client.<anonymous> (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/main.js:240:63)
    at Client.emit (node:events:513:28)
    at InteractionCreateAction.handle (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\actions\InteractionCreate.js:81:12)
    at module.exports [as INTERACTION_CREATE] (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\handlers\INTERACTION_CREATE.js:4:36)
    at WebSocketManager.handlePacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketManager.js:352:31)
    at WebSocketShard.onPacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:481:22)
    at WebSocketShard.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:321:10)
    at WebSocket.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\ws\lib\event-target.js:199:18)
2022-10-08T23:06:31.354Z [Quaver] INFO: Logging additional output to error.log.
```

### PR DESCRIPTION
**Pull request by zapteryx**
### Thank you for your interest in contributing!
Before proceeding, please review the [guidelines for contributing](https://github.com/ZPTXDev/Quaver/blob/master/CONTRIBUTING.md).

- [x] Are you targeting the `next` branch? (left side)
- [x] Did you review the guidelines for contributing?
- [x] Does your code pass linting checks?
- [x] Is your code documented, if applicable? (Check if not applicable)
- [x] Does this PR have a linked issue?

### Scope of change
- [x] Major change
- [ ] Minor change
- [ ] Documentation only

### Type of change
- [ ] Bug fix
- [x] Feature
- [ ] Other

### Priority
- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

### Description
Please describe the changes.
Fixes #524
Fixes #527
Closes #521

### CODE REVIEW

**Review by Xayanide — COMMENTED** [review]
dist folder should not be parsed by ESLint

**Review by Xayanide — CHANGES_REQUESTED** [review]
`dist` folder should be ignored by ESLint

**Review by Xayanide — CHANGES_REQUESTED** [review]
On the `settings.example`

```diff
// The database to use.
export const database = {
	// The path relative to the base directory. For example, database.sqlite would use the database.sqlite file.
-	path: 'database.sqlite',
+ 	path: '../database.sqlite',
};
```

Quaver will create a database.sqlite file inside the `dist` folder which we wouldn't want, so we must make it refer to the ones on the root outside the `src` or `dist` folder

**Review by Xayanide — CHANGES_REQUESTED** [review]
Submitting a modal causes a crash

```js
2022-10-08T10:14:53.758Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-08T10:14:53.762Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T10:14:53.764Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T10:14:53.766Z [Quaver] INFO: Disconnecting from all guilds...
2022-10-08T10:14:53.767Z [Quaver] INFO: [G 906471497231630336] Disconnecting (restarting)
2022-10-08T10:14:54.561Z [Quaver] ERROR: Cannot read properties of undefined (reading 'locale')
TypeError: Cannot read properties of undefined (reading 'locale')
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/components/modals/queue.js:19:44)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/events/interactionCreate.js:126:30)
    at Client.<anonymous> (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/main.js:239:63)
    at Client.emit (node:events:513:28)
    at InteractionCreateAction.handle (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\actions\InteractionCreate.js:81:12)
    at module.exports [as INTERACTION_CREATE] (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\handlers\INTERACTION_CREATE.js:4:36)     
    at WebSocketManager.handlePacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketManager.js:352:31)
    at WebSocketShard.onPacket (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:481:22)
    at WebSocketShard.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\client\websocket\WebSocketShard.js:321:10)
    at WebSocket.onMessage (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\ws\lib\event-target.js:199:18)
2022-10-08T10:14:54.568Z [Quaver] INFO: Logging additional output to error.log.
```

**Review by Xayanide — CHANGES_REQUESTED** [review]
Submitting a modal while Quaver is disconnected still crashes it
```js
$ npm run start

> quaver@5.2.3-next.1 start
> node .

2022-10-08T12:38:27.409Z [Discord] INFO: Connected. Logged in as Aia#3990.
2022-10-08T12:38:27.411Z [Quaver] INFO: Running version 5.2.3-next.1. For help, see https://github.com/ZPTXDev/Quaver/issues.
2022-10-08T12:38:27.411Z [Quaver] WARN: You are running an unstable version of Quaver. Please report bugs using the link above, and note that features may change or be removed entirely prior to release.
2022-10-08T12:38:28.177Z [Lavalink] INFO: Connected.
2022-10-08T12:38:29.951Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing modal queue_goto
2022-10-08T12:38:29.952Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-08T12:38:30.360Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T12:38:30.361Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T12:38:30.362Z [Quaver] ERROR: Unknown Message
DiscordAPIError[10008]: Unknown Message
    at SequentialHandler.runRequest (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:743:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async SequentialHandler.queueRequest (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:549:14)
    at async REST.request (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:988:22)
    at async MessageManager.delete (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\managers\MessageManager.js:254:5)
    at async Message.delete (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\structures\Message.js:762:5)
    at async Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/components/modals/queue.js:15:13)
2022-10-08T12:38:30.363Z [Quaver] INFO: Logging additional output to error.log.
```

File: src/components/buttons/search.ts [hunk 1]
```diff
@@ -39,27 +61,26 @@ export default {
...
 			else {
 				msg = 'MUSIC.QUEUE.TRACK_ADDED.MULTIPLE.DEFAULT';
-				extras = [resolvedTracks.length, await getGuildLocaleString(interaction.guildId, 'MISC.YOUR_SEARCH'), ''] ;
+				extras = [resolvedTracks.length.toString(), await getGuildLocaleString(interaction.guildId, 'MISC.YOUR_SEARCH'), ''] ;
 			}
 			if (!player?.connected) {
 				player = interaction.client.music.createPlayer(interaction.guildId);
 				player.handler = new PlayerHandler(interaction.client, player);
-				player.queue.channel = interaction.channel;
+				player.queue.channel = <TextChannel | VoiceChannel> interaction.channel;
```
Comment (Xayanide) [inline src/components/buttons/search.ts:69]: Based on usage, this is a TextChannel
```suggestion
				player.queue.channel = <TextChannel> interaction.channel;
```

File: src/components/buttons/search.ts [hunk 2]
```diff
@@ -39,27 +61,26 @@ export default {
...
 				await player.connect(interaction.member.voice.channelId, { deafened: true });
 				// Ensure that Quaver destroys the player if the user leaves the channel while Quaver is queuing tracks
 				// Ensure that Quaver destroys the player if Quaver gets timed out by the user while Quaver is queuing tracks
 				// Ensure that Quaver destroys the player if Quaver gets kicked or banned by the user while Quaver is queuing tracks
-				const timedOut = interaction.guild?.members.me.isCommunicationDisabled();
+				const me = await interaction.guild?.members.fetchMe();
+				const timedOut = me.isCommunicationDisabled();
```
Comment (Xayanide) [inline src/components/buttons/search.ts:75]: Can we do this same optimization for the if statement check cases on the other files involving us to check for timedOut?

File: src/lib/util/util.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	let embeds: EmbedBuilder[] = [];
+	const actionRow = new ActionRowBuilder<MessageActionRowComponentBuilder>();
+	switch (option) {
+		case 'language':
+			current = `${languageName[guildLocale] ?? 'Unknown'} (${guildLocale})`;
+			actionRow.addComponents(
+				new SelectMenuBuilder()
+					.setCustomId('language')
+					.addOptions(
+						fs.readdirSync(getAbsoluteFileURL(import.meta.url, ['..', '..', 'locales']))
```
Comment (Xayanide) [inline src/lib/util/util.ts:256]:
```js
2022-10-08T13:28:26.231Z [Quaver] INFO: Logging additional output to error.log.

Ava@DESKTOP-5531PRA MINGW64 ~/Documents/GitHub/Quaver (feat/transition-to-typescript)
$ 2022-10-08T13:28:24.967Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command settings
2022-10-08T13:28:24.968Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command settings
2022-10-08T13:28:24.973Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T13:28:24.974Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T13:28:24.975Z [Quaver] INFO: Disconnecting from all guilds...
2022-10-08T13:28:24.975Z [Quaver] INFO: [G 906471497231630336] Disconnecting (restarting)
2022-10-08T13:28:26.231Z [Quaver] ERROR: ENOENT: no such file or directory, scandir 'C:\Users\Ava\Documents\GitHub\Quaver\dist\locales'
Error: ENOENT: no such file or directory, scandir 'C:\Users\Ava\Documents\GitHub\Quaver\dist\locales'
    at Module.readdirSync (node:fs:1450:3)
    at settingsPage (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/lib/util/util.js:259:32)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/commands/settings.js:19:54)
2022-10-08T13:28:26.231Z [Quaver] INFO: Logging additional output to error.log.^C
```

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		"alwaysStrict": true,
+		"noEmitOnError": true,
+		"noImplicitAny": true,
+		"moduleResolution": "Node16",
+		"skipLibCheck": true,
+		"paths": {
+			"#src/*": ["src/*"]
+		}
+	},
+	"include": ["src/**/*"]
```
Comment (Xayanide) [inline tsconfig.json:18]: Not sure if we have to explicitly put an `"exclude": [""]` to exclude `node_modules`, `dist`, and so on, I guess this is fine as is.

**Review by Xayanide — CHANGES_REQUESTED** [review]

File: src/components/buttons/search.ts (hunk 1 above)
Comment (zapteryx) [reply src/components/buttons/search.ts:69]: could be a voicechannel due to text in voice

File: src/components/buttons/search.ts (hunk 2 above)
Comment (zapteryx) [reply src/components/buttons/search.ts:75]: sorry could you rephrase that

File: src/lib/util/util.ts (hunk 3 above)
Comment (zapteryx) [reply src/lib/util/util.ts:256]: whoops forgot to add another line

**Review by Xayanide — CHANGES_REQUESTED** [review]
Submitting the same page to queue throws an error e.g. Page 1 of 1, then attempting to go to page 1. Anything else makes it send the ephemeral Invalid input toast to the user which is expected.
```
2022-10-08T14:13:52.910Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing modal queue_goto
2022-10-08T14:13:52.910Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-08T14:13:52.914Z [Quaver] ERROR: The reply to this interaction has not been sent or deferred.
Error [InteractionNotReplied]: The reply to this interaction has not been sent or deferred.
    at ModalSubmitInteraction.editReply (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\structures\interfaces\InteractionResponses.js:150:48)
    at ReplyHandler.reply (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/lib/ReplyHandler.js:45:43)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/components/modals/queue.js:51:40)
```

**Review by Xayanide — CHANGES_REQUESTED** [review]
Submitting a modal while Quaver is disconnected still crashes it
```js
$ node .
2022-10-08T14:15:23.257Z [Discord] INFO: Connected. Logged in as Aia#3990.
2022-10-08T14:15:23.263Z [Quaver] INFO: Running version 5.2.3-next.1. For help, see https://github.com/ZPTXDev/Quaver/issues.
2022-10-08T14:15:23.263Z [Quaver] WARN: You are running an unstable version of Quaver. Please report bugs using the link above, and note that f
eatures may change or be removed entirely prior to release.
2022-10-08T14:15:23.985Z [Lavalink] INFO: Connected.
2022-10-08T14:15:25.786Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing modal queue_goto
2022-10-08T14:15:25.806Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing modal queue_goto
2022-10-08T14:15:26.489Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T14:15:26.490Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T14:15:26.491Z [Quaver] ERROR: Unknown Message
DiscordAPIError[10008]: Unknown Message
    at SequentialHandler.runRequest (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:743:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async SequentialHandler.queueRequest (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:549:14)
    at async REST.request (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\@discordjs\rest\dist\index.js:988:22)
    at async MessageManager.delete (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\managers\MessageManager.js:254:5)
    at async Message.delete (C:\Users\Ava\Documents\GitHub\Quaver\node_modules\discord.js\src\structures\Message.js:762:5)
    at async Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/components/modals/queue.js:15:13)
2022-10-08T14:15:26.491Z [Quaver] INFO: Logging additional output to error.log.
```

**Review by Xayanide — CHANGES_REQUESTED** [review]
Inside the `deploy-commands.js` script, `readdirSync` for commands should also be pointing at `dist`

**Review by Xayanide — COMMENTED** [review]
Everything inside the `main.ts` that has
```
file.endsWith('.js')
```
Can be this for debugging purposes.
```
- file.endsWith('.js')
+ file.endsWith('.js') || file.endsWith('.ts')
```

File: src/components/buttons/search.ts (hunk 2 above)
Comment (zapteryx) [reply src/components/buttons/search.ts:75]: oh sure

File: package.json [hunk 5]
```diff
@@ -19,24 +19,35 @@
...
+    "eslint": "^8.24.0",
+    "rimraf": "^3.0.2",
+    "tsc-alias": "^1.7.0",
+    "tsconfig-paths": "^4.1.0",
+    "typescript": "^4.8.4"
   },
-  "main": "src/main.js",
+  "main": "dist/main.js",
   "scripts": {
+    "rimraf": "./node_modules/rimraf/bin.js",
```
Comment (Xayanide) [inline package.json:39]: Using this script by itself doesn't work, it should be
```diff
-    "rimraf": "./node_modules/rimraf/bin.js",
+    "rimraf": "node ./node_modules/rimraf/bin.js",
```
Either way this is not needed the way I see it used.

File: package.json [hunk 6]
```diff
@@ -19,24 +19,35 @@
     "lavaclient": "^4.1.1",
     "lodash-es": "^4.17.21",
     "patch-package": "^6.4.7",
+    "remove": "^0.1.5",
```
Comment (Xayanide) [inline package.json:22]: Where is this dependency used?

**Review by Xayanide — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (Xayanide) [reply package.json]: I have gone and applied the former.

**Review by Xayanide — DISMISSED** [review]

File: src/lib/util/util.ts [hunk 7]
```diff
@@ -0,0 +1,294 @@
+import { resolve, dirname } from 'path';
+import { fileURLToPath, pathToFileURL } from 'url';
+import { get } from 'lodash-es';
+import { data, locales } from './common.js';
+import { colors, defaultLocaleCode } from '#src/settings.js';
+import { ActionRowBuilder, APISelectMenuOption, AttachmentBuilder, ButtonBuilder, ButtonStyle, ColorResolvable, EmbedBuilder, escapeMarkdown, Interaction, InteractionReplyOptions, MessageActionRowComponentBuilder, MessageCreateOptions, SelectMenuBuilder } from 'discord.js';
+import * as fs from 'fs';
```
Comment (Xayanide) [inline src/lib/util/util.ts:7]: What's wrong with using a named export here? Why `import *`?
```diff
- import * as fs from 'fs';
+ import { readdirSync } as fs from 'fs';
```

**Review by Xayanide — CHANGES_REQUESTED** [review]

File: src/lib/util/util.ts (hunk 7 above)
Comment (Xayanide) [reply src/lib/util/util.ts:7]: Have gone and applied the named export.

**Comment by Xayanide** [comment]
I will not be tackling all typing issues for this pr.

**Review by Xayanide — APPROVED** [review]

**Review by Xayanide — APPROVED** [review]
